components/reusable-components/navigation/Navigation.tsx

```diff
@@ -1,36 +1,38 @@
-/* eslint-disable jsx-a11y/no-static-element-interactions */
-/* eslint-disable jsx-a11y/click-events-have-key-events */
-
 'use client';
 
 import React, { useState, useRef } from 'react';
 import Link from 'next/link';
 import Image from 'next/image';
 import { usePathname } from 'next/navigation';
+import { useSession } from 'next-auth/react';
 import { FaUser, FaSignOutAlt, FaChevronDown, FaChevronUp, FaEnvelope } from 'react-icons/fa';
+import { Session } from 'next-auth';
 import sunImage from '../../../public/theme/sun.png';
 import moonImage from '../../../public/theme/moon.png';
 import LogoNavigation from '../../../public/logo/logo-black.svg';
 import styles from './navigation.module.scss';
 import { useTheme } from '../../../app/context/themeContext';
-import { User } from '../_Types';
+import { User, UserStatus } from '../_Types';
 import useClickOutside from '../../../api/utils/HOC/useClickOutside';
 import LogoutButton from '../button/LogoutButton';
 
-const user: User = {
-  email: 'john@example.com',
-  first_name: 'John',
-  last_name: 'Doe',
-  status: 'active',
-  image: null,
-};
+function extractUserFromSession(session: Session | null): User {
+  return {
+    email: typeof session?.user?.email === 'string' ? session.user.email : '',
```

Comment:
what other type session?.user?.email can be?
we should also handle null and invalid sessions by throwing an error for invalid or missing session data to catch problems earlier.